internal/cephfs/controllerserver.go

```diff
@@ -689,6 +689,22 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 			}
 			return &csi.DeleteSnapshotResponse{}, nil
 		}
+		// if the error is ErrVolumeNotFound, delete the snapshot as the
+		// snapshot and subvolume is already deleted
```

Comment:
If subvolume is deleted I expect the snapshots taken from the volume also does not exist, in that case, deleting a snapshot from the backend does not add value I believe. We have to correct the omap or journal entries though.

Reply:
Yes True. we just need to correct the omap entries. I will update the PR.

Reply:
@yati1998 yeah, just omap cleanup is enough :+1: